WinApi.Windows.Controls/Clipboard/Clipboard.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WinApi.Gdiplus;
+using WinApi.User32;
+
+namespace WinApi.Windows.Controls.Clipboard
+{
+    public  class Clipboard : IClipboard
+    {
+        private readonly CancellationTokenSource _cancellationTokenSource;
+        private readonly Task _waitAction;
+        private readonly ClipboardFormat[] _clipboardFormats;
+        private DummyForm _wind;
+
+        public Clipboard()
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _waitAction = Task.Run(() => Bootstrap(), _cancellationTokenSource.Token);
+            _clipboardFormats = new[] { ClipboardFormat.CF_UNICODETEXT, ClipboardFormat.CF_TEXT, ClipboardFormat.CF_BITMAP};
+        }
+
+        private void Bootstrap()
+        {
+            using (_wind = Window.Create<DummyForm>("hidden"))
+            {
+                _wind.ClipboardUpdateCallback = async () => await Update();
+
+                if (User32Methods.AddClipboardFormatListener(_wind.Handle))
+                    new EventLoop().Run(_wind);
+                else
+                {
+                    throw new Exception("Can't subscribe to the clipboard");
+                }
+            }
+        }
+
+        private void ReleaseUnmanagedResources()
+        {
+            //TODO: implement cancelation token
+            _cancellationTokenSource.Cancel();
```

Comment:
should probably somehow interact with the EventLoop().Run() method, just to stop it.